posthog/hogql/transforms/lazy_tables.py

```diff
@@ -206,6 +211,18 @@ def visit_select_query(self, node: ast.SelectQuery):
                 join_scope.from_table, join_scope.to_table, join_scope.fields_accessed, self.context.modifiers
             )
             join_to_add = cast(ast.JoinExpr, resolve_types(join_to_add, self.context, [node.type]))
+
+            # Hack to copy the events where conditions onto the session duration join
+            if isinstance(join_scope.lazy_join.join_table, EventsSessionSubTable):
+                if join_to_add.table.where is not None:
+                    self.compare_operators = []
+                    if node.where is not None:
+                        super().visit(node.where)
+                    super().visit(join_to_add.table.where)
+                    join_to_add.table.where = ast.And(exprs=self.compare_operators)
+                else:
+                    join_to_add.table.where = node.where
```

Comment:
I'm not super happy with this, feels hacky in its current form to copy over _all_ the `where` clauses. I couldn't think of a decent way to only pull out compare operators on certain fields (e.g. timestamp) due to the nature of `ast.Expr`